openedx/features/course_experience/static/course_experience/js/CourseHome.js

```diff
@@ -26,5 +29,97 @@ export class CourseHome {  // eslint-disable-line import/prefer-default-export
         },
       );
     });
+
+    $(document).ready(() => {
+      this.configureUpgradeMessage();
+    });
+  }
+
+  static fireSegmentEvent(event, properties) {
+    /* istanbul ignore next */
+    if (!window.analytics) {
+      return;
+    }
+
+    window.analytics.track(event, properties);
+  }
+
+  /**
+   * Persists the collapsed state of the upgrade message. If the message is collapsed,
```

Comment:
I'm wondering why this is persisted to local storage. Wouldn't the user expect it to be persisted across all their browsers?

Reply:
That would be nice, but persisting this across browsers is more work than we want to do. Also, we don't want to deal with the data.